src/bin/dump-db.rs

```diff
@@ -0,0 +1,91 @@
+//! Helper to set up the database role and permissions for creating database
+//! dumps using the `dump_db` task.
+//!
+//!  1. Set the `DUMP_DATABASE_URL` environment variable, e.g. by including it
+//!     in the `.env` file.
+//!
+//!  2. Run `dump-db setup` to create the database role.
+//!
+//!  3. Run `enqueue-job dump_db` to trigger a database dump.
+#![deny(warnings, clippy::all, rust_2018_idioms)]
+
+use diesel::connection::SimpleConnection;
+use diesel::prelude::*;
+
+use cargo_registry::{
+    db, env,
+    util::{human, CargoResult},
+};
+
+fn main() {
+    if let Err(e) = run() {
+        eprintln!("{}", e);
+        std::process::exit(1);
+    } else {
+        println!("Database updated successfully.");
+    }
+}
+
+fn run() -> CargoResult<()> {
+    let conn = db::connect_now()?;
+    let mut args = std::env::args().skip(1);
+    match args.next().as_ref().map(String::as_str) {
+        Some("setup") => db_setup(&conn),
+        Some("teardown") => db_teardown(&conn),
+        Some(_) | None => Err(human("unknown command")),
+    }
+}
+
+fn get_db_dump_credentials() -> CargoResult<(String, String)> {
+    let database_url = url::Url::parse(&env("DUMP_DATABASE_URL"))?;
```

Comment:
Can you clarify why we need a separate env var from `DATABASE_URL` for this?

Reply:
We need a different user name and password to allow us to use row-level security to filter rows. I found it easiest to use a different URL for this, since different authentication may also mean we need to use a different way of connecting to the DB. In my local setup, for example, I usually connect via the socket, but need to connect via TCP for DB users that are not tied to my system user, so the URL is different.